app/validators/orders_controller.rb

```diff
@@ -0,0 +1,63 @@
+module Validators
+  class OrdersController
+    include ActiveModel::Validations
+
+    attr_accessor :from_date, :to_date
+
+    validates :from_date, :to_date, presence: true
+    validate :params_are_permitted
+    validate :valid_date_format
+    validate :valid_date_range
```

Comment:
See [above](https://github.com/Ty0222/store-api/pull/1/files#r428297232) comment on why I went the route of leveraging `ActiveModel::Validations` for validating the `OrdersController` params.